processing/src/main/java/io/druid/segment/DimensionMerger.java

```diff
@@ -0,0 +1,165 @@
+{
+  /**
+   * Initialize any writer objects that can be instantiated before data from the segments has been processed/merged.
+   * @throws IOException
+   */
+  public void initWriters() throws IOException;
+
+
+  /**
+   * Read the segments to be merged and combine their indexes/data structures internally.
+   * (e.g., merging of dictionaries for String dimensions).
+   *
+   * @param adapters List of adapters to be merged.
+   * @throws IOException
+   */
+  public void mergeAcrossSegments(List<IndexableAdapter> adapters) throws IOException;
+
+
+  /**
+   * Update the DimensionMerger's internal structures with row value(s) to be merged.
+   *
+   * An example implementation might be to add the values to an open column writer.
+   *
+   * @param values The row values to be added.
+   * @throws IOException
+   */
+  public void addColumnValue(Object values) throws IOException;
+
+
+  /**
+   * Close any open writers and write merged data to disk.
+   *
+   * @param rowNumConversions A list of conversion buffers indexed by adapter number
+   *                          The list will have the same order as the list of adapters passed to mergeAcrossSegments()
+   *                          This buffer can be used to convert the row numbers from individual adapters.
+   * @throws IOException
+   */
+  public void closeWriters(List<IntBuffer> rowNumConversions) throws IOException;
+
+
+  /**
+   * When merging multiple segments, it is sometimes necessary to convert "missing" values to "nulls".
+   *
+   * Suppose we have String dimensions DimA and DimB and two segments to be merged.
+   *
+   * Segment 1 has rows:
+   *   {dimB = hello}
+   *
+   * Segment 2 has rows
+   *   {dimB = hello, dimA=null}
+   *   {dimB = world, dimA=foobar}
+   *
+   * When segment 1 and segment 2 are merged, dimA will not be present in segment 1 (no rows had values for dimA,
+   * so dimA was not written to disk with segment 1).
+   *
+   * For segment 2, dimA did have non-null values, so when it was persisted to disk, "null" would be
+   * assigned an encoded value (i.e., 0) within segment 2.
+   *
+   * The first row in both segments where dimB = hello should merge with each other.
+   *
+   * However, when rows from segment 1 and segment 2 are merged, the "missing" dimA values in segment 1 appear as
+   * null values within the merging process; the null dimA values from segment 2 appear as a dictionary encoding of 0.
+   *
+   * Thus, the rows from segment 1 and segment 2 where dimA is missing/null will not merge properly.
+   *
+   * To correct this, the missing values for dimA in segment 1 will be converted to
+   * encoded null values so that the rows can equate.
+   *
+   * @return true, if this conversion from missing->null is necessary
+   *         false, otherwise.
+   */
+  public boolean needsConvertMissingValues();
+
+
+  /**
+   * If a dimension has no values after merging across segments, it does not need to be written to disk and can be skipped.
```

Comment:
Errr, sorry, I didn't ask my question well.  If I have two segments, one has the dimension and the other doesn't, why will the merged output of those two segments not include the dimension.  The javadoc makes it sound like there are cases where the post-merge output might not contain a column that was in one of the inputs.